service/src/models/archaeologist.ts

```diff
@@ -153,29 +153,34 @@ export class Archaeologist {
   }
 
   async publishEnvConfig(connection) {
-    const envConfig = {
-      encryptionPublicKey: this.envConfig.encryptionPublicKey,
-      peerId: this.peerId.toString(),
-    };
-
-    const signature = await this.web3Interface.signer.signMessage(JSON.stringify(envConfig));
-
-    const configStr = JSON.stringify({
-      signature,
-      ...envConfig
-    });
-
-    const { stream } = await connection.newStream(`/env-config`);
-
-    pipe(
-      [new TextEncoder().encode(configStr)],
-      stream,
-      async (source) => {
-        for await (const data of source) {
-          const dataStr = new TextDecoder().decode(data as BufferSource | undefined);
-          console.log('dataStr', dataStr);
+    try {
+      const envConfig = {
+        encryptionPublicKey: this.envConfig.encryptionPublicKey,
+        peerId: this.peerId.toString(),
+      };
+
+      const signature = await this.web3Interface.signer.signMessage(JSON.stringify(envConfig));
+
+      const configStr = JSON.stringify({
+        signature,
+        ...envConfig
+      });
+
+      const { stream } = await connection.newStream(`/env-config`);
+
+      pipe(
+        [new TextEncoder().encode(configStr)],
+        stream,
+        async (source) => {
+          for await (const data of source) {
+            const dataStr = new TextDecoder().decode(data as BufferSource | undefined);
+            console.log('dataStr', dataStr);
+          }
         }
-      }
-    );
+      )
+    }
```

Comment:
This message is fired via a callback -- wondering if we could figure out a way to refactor this logic such that:

1. It gets called via the callback like it is now
2. The embalmer can also re-request this info without having to drop their connection and redial.

Thinking of the case where the connection succeeds, but for some reason they fail to get the arch response and want to re-request it.